Add CoCosNet model (#1542)
openvinotoolkit/open_model_zoo: demos/python_demos/cocosnet_demo/README.md

  -o OUTPUT_DIR, --output_dir OUTPUT_DIR
                        Optional. Path to directory to save the results
  -d DEVICE, --device DEVICE
                        Optional. Specify the target device to infer on; CPU,
                        GPU, FPGA, HDDL or MYRIAD is acceptable. Default
                        value is CPU
```

Running the application with the empty list of options yields the usage message given above and an error message.

To run the demo, you can use public or pre-trained models. You can download the pre-trained models with the OpenVINO [Model Downloader](../../../tools/downloader/README.md) or from [https://download.01.org/opencv/](https://download.01.org/opencv/).

> **NOTE**: Before running the demo with a trained model, make sure the model is converted to the Inference Engine format (\*.xml + \*.bin) using the [Model Optimizer tool](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html).

There are two ways to use this demo:

1. To use only correspondence and generative models (only CoCosNet). \
   In this case user have to set 3 inputs to the model (2 masks and 1 image).

>    You can use the following command do inference on CPU:

Reviewer: ```suggestion
   You can use the following command run demo on CPU:
```

   ```
       python3 cocosnet_demo.py

>        -c <path_to_corr_model>/Corr.xml

Reviewer: ```suggestion
       -d CPU
       -c <path_to_corr_model>/Corr.xml
```

       -g <path_to_gen_model>/Gen.xml
       -is <path_to_semantic_mask_of_image>/input_mask.png
       -ri <path_to_exemplar_image>/reference_image.jpg
       -rs <path_to_exemplar_semantic>/reference_mask.png
   ```

   > **NOTE**: For segmentation masks you should use mask (with shape: [height x width]) that specifies class for each pixel. Number of classes is 151 (from ADE20k), where '0' - nothing class.

2. To use the segmentation model in addition to CoCosNet. You should use only models trained on ADE20k dataset.     Example: [hrnet-v2-c1-segmentation](../../../models/public/hrnet-v2-c1-segmentation/hrnet-v2-c1-segmentation.md).
   In this case user have to set input image (.jpg) and reference image (.jpg) without any masks.
   Segmentation masks will be generated via segmentation model.

>    You can use the following command do inference on CPU:

Reviewer: ```suggestion
   You can use the following command run demo on CPU:
```

   ```
   python3 cocosnet_demo.py

>        -c <path_to_corr_model>/Corr.xml

Reviewer: ```suggestion
       -d CPU
       -c <path_to_corr_model>/Corr.xml
```

Running the application with the empty list of options yields the usage message given above and an error message.

To run the demo, you can use public or pre-trained models. You can download the pre-trained models with the OpenVINO [Model Downloader](../../../tools/downloader/README.md) or from [https://download.01.org/opencv/](https://download.01.org/opencv/).

> **NOTE**: Before running the demo with a trained model, make sure the model is converted to the Inference Engine format (\*.xml + \*.bin) using the [Model Optimizer tool](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html).

There are two ways to use this demo:

1. To use only correspondence and generative models (only CoCosNet). \
   In this case user have to set 3 inputs to the model (2 masks and 1 image).
   You can use the following command run demo on CPU:

   ```
   python3 cocosnet_demo.py \
       -d CPU \
       -c <path_to_corr_model>/Corr.xml \
       -g <path_to_gen_model>/Gen.xml \
       -is <path_to_semantic_mask_of_image>/input_mask.png \
       -ri <path_to_exemplar_image>/reference_image.jpg \
       -rs <path_to_exemplar_semantic>/reference_mask.png
   ```

>    > **NOTE**: For segmentation masks you should use mask (with shape: [height x width]) that specifies class for each pixel. Number of classes is 151 (from ADE20k), where '0' - nothing class.

Reviewer 2: ```suggestion
   > **NOTE**: For segmentation masks you should use mask (with shape: [height x width]) that specifies class for each pixel. Number of classes is 151 (from ADE20k), where '0' - background class.
```
